pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -45,36 +47,35 @@ const (
 )
 
 func AddToManager(mgr manager.Manager, architectures []string, dockerCfgPath string) error {
-	if err := buildsreconciler.AddToManager(mgr); err != nil {
-		return fmt.Errorf("failed to construct builds reconciler: %w", err)
-	}
-
 	logger := logrus.WithField("controller", controllerName)
-	c, err := controller.New(controllerName, mgr, controller.Options{
-		MaxConcurrentReconciles: 1,
-		Reconciler: &reconciler{
-			logger:         logger,
-			client:         mgr.GetClient(),
-			architectures:  architectures,
-			manifestPusher: manifestpusher.NewManifestPushfer(logger, registryURL, dockerCfgPath),
-			imageMirrorer:  &ocImage{log: logger, registryConfig: dockerCfgPath},
-			scheme:         mgr.GetScheme(),
-		},
-	})
-	if err != nil {
-		return fmt.Errorf("failed to construct controller: %w", err)
-	}
 
-	predicateFuncs := predicate.Funcs{
+	mabcPredicateFuncs := predicate.Funcs{
 		CreateFunc:  func(e event.CreateEvent) bool { return true },
 		DeleteFunc:  func(event.DeleteEvent) bool { return false },
 		UpdateFunc:  func(event.UpdateEvent) bool { return true },
 		GenericFunc: func(event.GenericEvent) bool { return false },
 	}
 
-	if err := c.Watch(source.Kind(mgr.GetCache(), &v1.MultiArchBuildConfig{}),
-		&handler.EnqueueRequestForObject{}, predicateFuncs); err != nil {
-		return fmt.Errorf("failed to create watch for MultiArchBuildConfig: %w", err)
+	buildPredicateFuncs := predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		DeleteFunc:  func(event.DeleteEvent) bool { return false },
+		UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+		GenericFunc: func(event.GenericEvent) bool { return false },
+	}
+
+	if err := ctrlruntime.NewControllerManagedBy(mgr).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
+		For(&v1.MultiArchBuildConfig{}, builder.WithPredicates(mabcPredicateFuncs)).
+		Owns(&buildv1.Build{}, builder.WithPredicates(buildPredicateFuncs)).
```

Comment:
We need only the builds that have been created by us with specific labels.

Reply:
It's true but the point is the reconciler, using this configuration, is going to enqueue not any build but only those which belong to an `mabc` we created (builds having the label we want), that's thanks to the `ownerReferences` we set and `Owns()`. I could add the extra check on the label but it won't filter anything I guess.

Reply:
That is fine. Since the reconciler will be triggered, why do we need to list the builds since the event was triggered for a specific build instead?

Reply:
Because the `mabc` owns those `builds` and it needs to check their state to decide how to move ahead.
As an example:
- one of the builds controlled by any `mabc`s fails
- the controller gets notified and enqueue a new request for the parent `mabc` and it calls `r.reconcile()` 
- the reconciler lists the builds, notices one failed and mark the `mabc` as failed too

Again, if your concern is about spamming, I'd reassure you saying that `r.client.List(ctx, &builds, &listOpts)` it's likely to hit the manager cache rather than the API Server.
Furthermore, we are going to ignore any build not related to an `mabc`, so we won't get notified every time.